### PR TITLE
[AptosFramework] fix the error code in epilogue

### PR DESCRIPTION
**Pull request by zekun000**
The epilogue error conversion requires the exact code for not enough gas to charge properly, otherwise it's failed with UNEXPECTED_ERROR_FROM_KNOWN_MOVE_FUNCTION and being discarded.

### CODE REVIEW

File: testsuite/smoke-test/src/aptos/gas_check.rs [hunk 1]
```diff
@@ -35,6 +35,17 @@ impl AptosTest for GasCheck {
...
+
+        let transfer_too_much =
+            account2.sign_with_transaction_builder(ctx.aptos_transaction_factory().payload(
+                aptos_stdlib::encode_test_coin_transfer(account1.address(), 1000),
+            ));
+
+        ctx.client()
+            .submit_and_wait(&transfer_too_much)
+            .await
+            .unwrap_err();
```
Comment (msmouse) [inline testsuite/smoke-test/src/aptos/gas_check.rs:48]: Does this guarantee it's an error instead of a discard?

Comment (zekun000) [reply testsuite/smoke-test/src/aptos/gas_check.rs:48]: discard will timeout this request and probably timeout the whole test

Comment (msmouse) [reply testsuite/smoke-test/src/aptos/gas_check.rs:48]: K. Would be nice if we can test the explicit error. But this works for me.

**Review by msmouse — APPROVED** [review]

**Pull request closed without merging**